src/App.tsx

```diff
@@ -32,64 +33,128 @@ export function getReorderedGoods(
   { sortType, isReversed }: ReorderOptions,
 ) {
   // To avoid the original array mutation
-  const visibleGoods = [...goods];
+  let visibleGoods = [...goods];
+
+  if (sortType === SortType.ALPHABET) {
+    visibleGoods = [...visibleGoods].sort((a, b) => a.localeCompare(b));
+  } else if (sortType === SortType.LENGTH) {
+    visibleGoods = [...visibleGoods].sort((a, b) => a.length - b.length);
+  }
+
+  if (isReversed) {
+    visibleGoods = [...visibleGoods].reverse();
+  }
```

Comment:
your visibleGoods is already a copy of goods, so it's unnecessary to copy it once more
it woukd be enough: 
``` 
if (sortType === SortType.ALPHABET) {
    visibleGoods.sort((a, b) => a.localeCompare(b));
}
```